authLdap.php

```diff
@@ -254,10 +261,17 @@ function authLdap_login($user, $username, $password, $already_md5 = false)
             return false;
         }
 
-        $role = '';
+        // if the user exists, use that role
+        $uid = authLdap_get_uid($username);
+        $role = authLdap_user_role($uid);
+
         // do some group mapping
         if ($authLDAPGroupEnable) {
```

Comment:
When we use LDAP-based roles we **have** to overwrite any changes that have been added to wordpress. Changes shall be made in LDAP. But when we do **not** use the LDAP-based roles it is futile to overwrite changes made in wordpress. Therefore  - in my opinion - this should read ```if (! $authLDAPGroupEnable) { ```. Or am I (again :wink:) missing something?